File: app/src/main/java/com/icostel/arhitecturesample/api/Status.java
package com.icostel.arhitecturesample.api;

public class Status {
    public enum Type {NOT_STARTED, IN_PROGRESS, SUCCESS, ERROR, LOADING, CALL_ERROR, INPUTS_ERROR}

    private Type type;

    private Status(Type type) {
        this.type = type;
    }

    public static com.icostel.arhitecturesample.api.Status NotStarted() {
        return new Status(Type.NOT_STARTED);
    }

    public static com.icostel.arhitecturesample.api.Status InProgress() {
        return new Status(Type.IN_PROGRESS);
    }

    public static com.icostel.arhitecturesample.api.Status Success() {
        return new Status(Type.SUCCESS);
    }

    public static com.icostel.arhitecturesample.api.Status Error() {
        return new Status(Type.CALL_ERROR);
    }

    public static com.icostel.arhitecturesample.api.Status Loading() {
        return new Status(Type.LOADING);
    }

    public Type getType() {
        return this.type;
    }
}

File: app/src/main/java/com/icostel/arhitecturesample/api/Status.kt
package com.icostel.arhitecturesample.api

class Status private constructor(val type: Type) {
    enum class Type {
        NOT_STARTED, SUCCESS, ERROR, LOADING, CALL_ERROR, INPUTS_ERROR
    }

    companion object {

        fun notStarted(): Status {
            return Status(Type.NOT_STARTED)
        }

        fun success(): Status {
            return Status(Type.SUCCESS)
        }

        fun error(): Status {
            return Status(Type.CALL_ERROR)
        }

        fun loading(): Status {
            return Status(Type.LOADING)
        }
    }
}
